<p>There are valid cases for passing a variable multiple times into the same method call, but usually doing so is a mistake, and something else was
intended for one of the arguments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
If compare(myPoint.x, myPoint.x) &lt;&gt; 0 Then ' Noncompliant
    '...
End If

If compare(getNextValue(), getNextValue()) &lt;&gt; 0 Then ' Noncompliant
    '...
End If
</pre>
<h2>Compliant Solution</h2>
<pre>
If compare(myPoint.x, myPoint.y) &lt;&gt; 0 Then
    '...
End If

Dim v1 As Integer = getNextValue()
Dim v2 As Integer = getNextValue()
If compare(v1, v2) &lt;&gt; 0 Then
    '...
End If
</pre>
<h2>Deprecated</h2>
<p>This rule is deprecated, and will eventually be removed.</p>

